src/rust/ide/lib/view/src/graph_editor/component/breadcrumbs.rs

```diff
@@ -0,0 +1,366 @@
+            }
+            self.current_index.set(next_index);
+            sources.append(&mut self.update_selection());
+        }
+        sources
+    }
+
+    fn pop_breadcrumb(&self) -> Vec<frp::Source> {
+        debug!(self.logger, "Popping {self.current_index.get()}");
+        if self.current_index.get() > 0 {
+            info!(self.logger, "Popping breadcrumb view.");
+            self.current_index.set(self.current_index.get() - 1);
+            self.update_selection()
+        } else {
+            default()
+        }
+    }
+
+    fn update_selection(&self) -> Vec<frp::Source> {
+        let mut sources = Vec::new();
+        let current_index = self.current_index.get();
+        for (index,breadcrumb) in self.breadcrumbs.borrow_mut().iter().enumerate() {
+            if index + 1 == current_index {
+                sources.push(breadcrumb.frp.select.clone_ref());
+            } else {
+                sources.push(breadcrumb.frp.deselect.clone_ref());
+            }
+        }
+        sources
+    }
+
+    fn remove_breadcrumbs_history(&self, index:usize) {
+        for breadcrumb in self.breadcrumbs.borrow_mut().split_off(index) {
+            info!(self.logger, "Removing {breadcrumb.info.method_pointer.name}.");
+            breadcrumb.unset_parent();
+        }
+    }
+}
+
+impl display::Object for BreadcrumbsModel {
+    fn display_object(&self) -> &display::object::Instance {
+        &self.display_object
+    }
+}
+
+
+
+// ===================
+// === Breadcrumbs ===
+// ===================
+
+/// The Breadcrumb's view used for visualizing the breadcrumbs and navigating them.
+#[derive(Debug,Clone,CloneRef,Shrinkwrap)]
+#[allow(missing_docs)]
+pub struct Breadcrumbs {
+    #[shrinkwrap(main_field)]
+    model   : Rc<BreadcrumbsModel>,
+    pub frp : Frp
+}
+
+impl Breadcrumbs {
+    /// Constructor.
+    pub fn new<'t,S:Into<&'t Scene>>(scene:S, focus_manager:&FocusManager) -> Self {
+        let frp     = Frp::new();
+        let model   = Rc::new(BreadcrumbsModel::new(scene,&frp,focus_manager));
+        let network = &frp.network;
+        frp::extend! { network
+            eval frp.push_breadcrumb((local_call) {
+                model.push_breadcrumb(local_call).iter().for_each(|source| source.emit(()));
+            });
+            eval_ frp.pop_breadcrumb(
+                model.pop_breadcrumb().iter().for_each(|source| source.emit(()))
+            );
+        }
+
+
+        // === GUI Update ===
+
+        frp::extend! { network
+            eval model.project_name.frp.outputs.width((width) {
+                model.relayout_for_project_name_width(*width)
+            });
+        }
+
+
+        // === User Interaction ===
+
+        frp::extend! {network
+            eval_ model.project_name.frp.outputs.mouse_down(
+                model.select_breadcrumb(0).iter().for_each(|source| source.emit(()))
+            );
+        }
+
+
+        Self{frp,model}
```

Comment:
Here should be the whole logic of the FRP system. So, after you call `select_breadcrumb` you should get list of breadcrumbs to be popped etc and here you should handle it. This is the part where the "controller" should be implemented